### PR TITLE
adding highlight feature and select lines feature

### PR DESCRIPTION
**Pull request by flossypurse**
This PR adds two new "per snip" features.

**Selected lines**

A single source code snippet may be used in multiple places.
If so, you can  customize which lines are rendered.
Add a "selected" configuration to the snip start line.

```
<!--SNIPSTART hellouniverse {"selectedLines": ["1", "3-5"]}-->
```

The feature supports multiple selections as either a single line or a range.

**Highlighed lines**

Some frameworks support highlighting code lines in code blocks (such as [Docusaurus](https://docusaurus.io/docs/markdown-features/code-blocks#line-highlighting). 
If so, you can add a "highlights" configuration to the snip start line.

```
<!--SNIPSTART hellouniverse {"highlightedLines": "{1, 3-4}"}-->
```

### CODE REVIEW

**Comment by bergundy** [comment]
I wouldn't reference line numbers in the snippet config but rather do that in the original code.
Line numbers tend to change and will break the docs once they do.

For highlights I'd rather have this:
```
// @@@SNIPSTART snip-name

some code ...

// @@@SNIPHIGHLIGHTSTART
some highlighted code
// @@@SNIPHIGHLIGHTEND

...

// @@@SNIPEND

```

For selecting lines, this might be an option:
```
// @@@SNIPSTART snip-name
some code...
// @@@SNIPGROUPSTART group-name
some selected code..
// @@@SNIPGROUPEND group-name
some unselected code..
// @@@SNIPGROUPSTART group-name
some more selected code..
// @@@SNIPGROUPEND group-name
// @@@SNIPEND
```

Then in the snip config use:
```
<!--SNIPSTART hellouniverse {"group": "group-name"}-->
```

**Comment by flossypurse** [comment]
> I wouldn't reference line numbers in the snippet config but rather do that in the original code.

Yes, however I want to be able to use the same snippet in multiple places, but select or highlight different lines.
By putting that in the source code, it defeats that purpose.

> Line numbers tend to change and will break the docs once they do.

Yes, it would be a carefully considered use case. 

In theory, we could support both ways.

**Comment by bergundy** [comment]
What about only having named groups and in the snip config have:
```markdown
<!--SNIPSTART hellouniverse {"select": ["group-name"], "highlight": ["group-a", "group-b"]}-->
```

That gives you all of the functionality you need without being dependent on line numbers.

File: README.md [hunk 1]
```diff
@@ -112,6 +112,34 @@ To customize how a single snip is rendered, add a JSON feature configuration in
...
+
+The feature supports multiple selections as either a single line or a range.
+
+**Highlighed lines**
+
+Some frameworks support highlighting code lines in code blocks.
+If so, you can add a "highlights" configuration to the snip start line.
+
+```
+<!--SNIPSTART hellouniverse {"highlights": "{1, 3-4}"}-->
```
Comment (flossypurse) [inline README.md:135]:
```suggestion
<!--SNIPSTART hellouniverse {"highlightedLines": "{1, 3-4}"}-->
```

**Review by napcs — APPROVED** [review]

**Pull request merged**